cli/cmd/helm/helmInstall.go

```diff
@@ -0,0 +1,202 @@
+	if err != nil {
+		return fmt.Errorf("could not install The Observability Stack: %w", err)
+	}
+
+	keyFile, err := cmd.Flags().GetString("timescaledb-tls-key")
+	if err != nil {
+		return fmt.Errorf("could not install The Observability Stack: %w", err)
+	}
+
+	if certFile != "" && keyFile != "" {
+		i.tsDBTlsCert, err = ioutil.ReadFile(certFile)
+		if err != nil {
+			return fmt.Errorf("failed to read file %s: %v", certFile, err)
+		}
+
+		i.tsDBTlsKey, err = ioutil.ReadFile(keyFile)
+		if err != nil {
+			return fmt.Errorf("failed to read file %s: %v", keyFile, err)
+		}
+	}
+
+	err = i.installStack()
+	if err != nil {
+		return fmt.Errorf("could not install The Observability Stack: %w", err)
+	}
+	return nil
+}
+
+func (c *installSpec) installStack() error {
+	var err error
+	// if custom helm chart is provided there is no point
+	// of adding & upgrading the default tobs helm chart
+	if c.ref == utils.DEFAULT_CHART {
+		err = utils.AddTobsHelmChart()
+		if err != nil {
+			return err
+		}
+
+		err = utils.UpdateTobsHelmChart(false)
+		if err != nil {
+			return err
+		}
+	}
+
+	cmds := []string{"install", cmd.HelmReleaseName, c.ref}
+
+	// Note do not change the below order the --set flag is set
+	// in above we are appending more flags to it with below check
+	helmValues := "cli=true"
+	if c.dbURI != "" {
+		helmValues = appendDBURIValues(c.dbURI, cmd.HelmReleaseName, helmValues)
+	}
+
+	// If enable backup is disabled by flag check the backup option
+	// from values.yaml as a second option
+	if !c.enableBackUp {
+		e, err := utils.ExportValuesFieldFromChart(c.ref, TimescaleDBBackUpKeyForValuesYaml)
+		if err != nil {
+			return err
+		}
+		var ok bool
+		c.enableBackUp, ok = e.(bool)
+		if !ok {
+			return fmt.Errorf("enable Backup was not a bool")
+		}
+	} else {
+		// update timescaleDB backup in values.yaml
+		helmValues = helmValues + ",timescaledb-single.backup.enabled=true"
+	}
+
+	t := timescaledb_secrets.TSDBSecretsInfo{
+		ReleaseName:    cmd.HelmReleaseName,
+		Namespace:      cmd.Namespace,
+		EnableS3Backup: c.enableBackUp,
+		TlsCert:        c.tsDBTlsCert,
+		TlsKey:         c.tsDBTlsKey,
+	}
+
+	// if db-uri is provided we do not need
+	// to create DB level secrets
+	if c.dbURI == "" {
+		err = t.CreateTimescaleDBSecrets()
+		if err != nil {
+			return err
+		}
+	}
+
+	if cmd.Namespace != "default" {
+		cmds = append(cmds, "--create-namespace", "--namespace", cmd.Namespace)
+	}
+	if c.configFile != "" {
+		cmds = append(cmds, "--values", c.configFile)
+	}
+	if DEVEL {
+		cmds = append(cmds, "--devel")
+	}
+
+	cmds = append(cmds, "--set", helmValues)
+	install := exec.Command("helm", cmds...)
+	fmt.Println("Installing The Observability Stack")
+	out, err := install.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("could not install The Observability Stack: %w \nOutput: %v", err, string(out))
+	}
+
+	time.Sleep(10 * time.Second)
+
+	fmt.Println("Waiting for pods to initialize...")
```

Comment:
Should this `Println` be before the `Sleep`?

Reply:
So the install includes two steps 
1. tobs install internally triggers helm install process. Here we are waiting `10 secs` after helm install to proceed with the step 2. 
2. Check for pods status. After the successful helm install. tobs waits for pods to get to running state by polling/waiting for pods to get into running state before the `tobs install` cmd exits. 

So the wait is to let `helm install` cmd to kick off. So that we can check the pods status that needs to be running after helm install.